modules/distributor/receiver/metrics_provider.go

```diff
@@ -22,184 +22,82 @@ import (
 	"github.com/prometheus/client_golang/prometheus/promauto"
 	"go.opentelemetry.io/otel/metric"
 	"go.opentelemetry.io/otel/metric/embedded"
+	"go.opentelemetry.io/otel/metric/noop"
 )
 
 var (
 	// Compile-time check this implements the OpenTelemetry API.
 
-	_ metric.MeterProvider                  = MeterProvider{}
-	_ metric.Meter                          = Meter{}
-	_ metric.Observer                       = Observer{}
-	_ metric.Registration                   = Registration{}
-	_ metric.Int64Counter                   = Int64Counter{}
-	_ metric.Float64Counter                 = Float64Counter{}
-	_ metric.Int64UpDownCounter             = Int64UpDownCounter{}
-	_ metric.Float64UpDownCounter           = Float64UpDownCounter{}
-	_ metric.Int64Histogram                 = Int64Histogram{}
-	_ metric.Float64Histogram               = Float64Histogram{}
-	_ metric.Int64Gauge                     = Int64Gauge{}
-	_ metric.Float64Gauge                   = Float64Gauge{}
-	_ metric.Int64ObservableCounter         = Int64ObservableCounter{}
-	_ metric.Float64ObservableCounter       = Float64ObservableCounter{}
-	_ metric.Int64ObservableGauge           = Int64ObservableGauge{}
-	_ metric.Float64ObservableGauge         = Float64ObservableGauge{}
-	_ metric.Int64ObservableUpDownCounter   = Int64ObservableUpDownCounter{}
-	_ metric.Float64ObservableUpDownCounter = Float64ObservableUpDownCounter{}
-	_ metric.Int64Observer                  = Int64Observer{}
-	_ metric.Float64Observer                = Float64Observer{}
+	_ metric.MeterProvider = &MeterProvider{}
```

Comment:
Do you know if we need the pointer? I think we can keep the value receivers, none of the methods modify the struct state

Reply:
Sure, I've pushed a new commit with just value receivers. Works just as well.